express-get-json/index.js

```diff
@@ -0,0 +1,48 @@
+const express = require('express');
+const app = express();
+const PORT = 3000;
+
+const grades = {
+  12: {
+    id: 12,
+    name: 'Tim Berners-Lee',
+    course: 'HTML',
+    score: 95
+  },
+  47: {
+    id: 47,
+    name: 'Brendan Eich',
+    course: 'JavaScript',
+    score: 100
+  },
+  273: {
+    id: 273,
+    name: 'Forbes Lindsay',
+    course: 'JavaScript',
+    score: 92
+  }
+};
+
+app.use('/api/grades', (req, res, next) => {
+  switch (req.method) {
+    case 'GET':
+      res.send(gradeArray(grades));
+      break;
+    default:
+      console.error(`unsupported request method ${req.method}`);
+      break;
+  }
+});
```

Comment:
Rather than testing the request method part-way through the middleware, use one of the specialize `app` methods so the middleware only triggers on requests with that method: http://expressjs.com/en/starter/basic-routing.html

Reply:
gotcha, thanks